docs/querying/api.md

```diff
@@ -673,7 +673,9 @@ GET /api/v1/rules
 ```
 
 URL query parameters:
+
 - `type=alert|record`: return only the alerting rules (e.g. `type=alert`) or the recording rules (e.g. `type=record`). When the parameter is absent or empty, no filtering is done.
+- `rules=alertName,RuleName`: return only the alerting and recording rules with the specified names. If we've filtered out all the rules of a group, the group is not returned. When the parameter is absent or empty, no filtering is done.
```

Comment:
Rule names may contain commas, so we would have to introduce escaping to make this fully correct. But I would propose just allowing the parameter to be repeated, like we do for the `match[]` parameter in various API endpoints (see https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers for example). That's both an established pattern and keeps things simpler.

The question is then whether `rule[]` would be a good name for the repeated filter parameter. Maybe `rule_name[]` to make it more explicit what we're filtering on, especially in case there's other future filters for this endpoint (e.g. I guess it would make sense to also filter by rule group name at the same time to fully scope your query and not accidentally select matching rule names from an unrelated group?).

Regarding naming I also noticed that we accidentally(?) started using both camelCase (e.g. `scrapePool` in the `/api/v1/targets` endpoint) and snake_case (e.g. `match_target` in `/api/v1/targets/metadata`) for HTTP parameters. But snake case was there first and is also more consistent with the HTTP path naming that we use for our endpoints, so I would prefer it for future parameter names I think.

Reply:
Good point about the file name! Yeah, that sounds great to me :)